Support commons-io 2.7

Released 28 May 2020: http://commons.apache.org/proper/commons-io/changes-report.html#a2.7

[ERROR] Failed to execute goal [32mde.thetaphi:forbiddenapis:3.0:check[m [1m(check)[m on project [36mmac-bc360-custom[m: [1;31mParsing signatures failed: Method not found while parsing signature: org.apache.commons.io.output.ByteArrayOutputStream#toString()[m -> [1m[Help 1][m

With the stack trace:
[ERROR] Failed to execute goal de.thetaphi:forbiddenapis:3.0:check (check) on project mac-bc360-custom: Parsing signatures failed: Method not found while parsing signature: org.apache.commons.io.output.ByteArrayOutputStream#toString() -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal de.thetaphi:forbiddenapis:3.0:check (check) on project mac-bc360-custom: Parsing signatures failed: Method not found while parsing signature: org.apache.commons.io.output.ByteArrayOutputStream#toString()
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Parsing signatures failed: Method not found while parsing signature: org.apache.commons.io.output.ByteArrayOutputStream#toString()
    at de.thetaphi.forbiddenapis.maven.AbstractCheckMojo.execute (AbstractCheckMojo.java:432)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: de.thetaphi.forbiddenapis.ParseException: Method not found while parsing signature: org.apache.commons.io.output.ByteArrayOutputStream#toString()
    at de.thetaphi.forbiddenapis.Signatures$UnresolvableReporting$1.parseFailed (Signatures.java:56)
    at de.thetaphi.forbiddenapis.Signatures.addSignature (Signatures.java:202)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesFile (Signatures.java:302)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesStream (Signatures.java:268)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:264)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesFile (Signatures.java:284)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesStream (Signatures.java:268)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:264)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesFile (Signatures.java:284)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesStream (Signatures.java:268)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:264)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesFile (Signatures.java:284)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesStream (Signatures.java:268)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:264)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesFile (Signatures.java:284)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesStream (Signatures.java:268)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:264)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesFile (Signatures.java:284)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesStream (Signatures.java:268)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:264)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesFile (Signatures.java:284)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesStream (Signatures.java:268)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:264)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesFile (Signatures.java:284)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesStream (Signatures.java:268)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:264)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesFile (Signatures.java:284)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesStream (Signatures.java:268)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:264)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesFile (Signatures.java:284)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesStream (Signatures.java:268)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:264)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesFile (Signatures.java:284)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesStream (Signatures.java:268)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:264)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesFile (Signatures.java:284)
    at de.thetaphi.forbiddenapis.Signatures.parseSignaturesStream (Signatures.java:268)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:264)
    at de.thetaphi.forbiddenapis.Signatures.addBundledSignatures (Signatures.java:311)
    at de.thetaphi.forbiddenapis.Checker.addBundledSignatures (Checker.java:317)
    at de.thetaphi.forbiddenapis.maven.AbstractCheckMojo.execute (AbstractCheckMojo.java:393)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)

--- a/src/main/resources/de/thetaphi/forbiddenapis/signatures/commons-io-unsafe-1.0.txt
+++ b/src/main/resources/de/thetaphi/forbiddenapis/signatures/commons-io-unsafe-1.0.txt
@@ -14,8 +14,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-@ignoreUnresolvable
-
 @defaultMessage Uses default charset
 org.apache.commons.io.CopyUtils#copy(byte[],java.io.Writer)
 org.apache.commons.io.CopyUtils#copy(java.io.InputStream,java.io.Writer)
@@ -25,4 +23,6 @@ org.apache.commons.io.IOUtils#toByteArray(java.io.Reader)
 org.apache.commons.io.IOUtils#toByteArray(java.lang.String)
 org.apache.commons.io.IOUtils#toString(byte[])
 org.apache.commons.io.IOUtils#toString(java.io.InputStream)
+
+@ignoreUnresolvable
 org.apache.commons.io.output.ByteArrayOutputStream#toString()

--- a/src/main/resources/de/thetaphi/forbiddenapis/signatures/commons-io-unsafe-2.7.txt
+++ b/src/main/resources/de/thetaphi/forbiddenapis/signatures/commons-io-unsafe-2.7.txt
@@ -14,15 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-@ignoreUnresolvable
+@includeBundled commons-io-unsafe-2.6
 
 @defaultMessage Uses default charset
-org.apache.commons.io.CopyUtils#copy(byte[],java.io.Writer)
-org.apache.commons.io.CopyUtils#copy(java.io.InputStream,java.io.Writer)
-org.apache.commons.io.CopyUtils#copy(java.io.Reader,java.io.OutputStream)
-org.apache.commons.io.CopyUtils#copy(java.lang.String,java.io.OutputStream)
-org.apache.commons.io.IOUtils#toByteArray(java.io.Reader)
-org.apache.commons.io.IOUtils#toByteArray(java.lang.String)
-org.apache.commons.io.IOUtils#toString(byte[])
-org.apache.commons.io.IOUtils#toString(java.io.InputStream)
-org.apache.commons.io.output.ByteArrayOutputStream#toString()
+org.apache.commons.io.output.AbstractByteArrayOutputStream#toString()

